plugin/formatting.py

```diff
@@ -97,13 +97,24 @@ def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
         session = self.best_session(self.capability)
         if session:
             # Either use the documentFormattingProvider ...
-            session.send_request(text_document_formatting(self.view), self.on_result)
+            req = text_document_formatting(self.view)
+
+            def run_async() -> None:
+                assert session  # TODO: How to make mypy shut up about an Optional[Session]?
```

Comment:
You could assign `session` to another variable. I think it might be complaining because it's assuming that something else might change the `session` variable in another branch.

Reply:
I have a plan to fix this in a neat way. These asserts are okay for now.